File: RunPS.py
import os,sys,clr

SMA_Dir = (lambda s: s + os.listdir(s)[0])(r'C:\\Windows\\Microsoft.NET\\assembly\GAC_MSIL\\System.Management.Automation\\')
sys.path.append(SMA_Dir)

clr.AddReference(r"C:\\Windows\\Microsoft.NET\\Framework64\\v4.0.30319\\System.Runtime.dll")
clr.AddReference("System.Management.Automation")
clr.AddReference('System.Collections')
clr.AddReference('System.Management')

from System import Guid
from System.Diagnostics.Eventing import EventProvider
from System.Management.Automation import RunspaceInvoke
from System.Management.Automation.Runspaces import RunspaceFactory
from System.Reflection import BindingFlags
from System.Text import StringBuilder


def run_ps(command):
        runspace = RunspaceFactory.CreateRunspace()
        runspace.Open()
        RunspaceInvoke(runspace)
        pipeline = runspace.CreatePipeline()
        pipeline.Commands.AddScript(command)
        pipeline.Commands.Add("Out-String")
        result = pipeline.Invoke()
        runspace.Close()
        stringBuilder = StringBuilder()
        for each in result:
            stringBuilder.Append(each)
        return stringBuilder.ToString()

if __name__ == '__main__':
   
    command = 'calc'
    output = run_ps(command)
    print(output)

File: C++/RunPS/RunPS.cpp
#include "pch.h"
#using <System.dll>
#using <C:\Windows\Microsoft.NET\assembly\GAC_MSIL\System.Management.Automation\v4.0_3.0.0.0__31bf3856ad364e35\System.Management.Automation.dll>
#using <C:\Windows\Microsoft.NET\assembly\GAC_MSIL\System.Collections\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Collections.dll>

using namespace System;
using namespace System::Collections::ObjectModel;
using namespace System::Management::Automation;
using namespace System::Management::Automation::Runspaces;

int main(array<System::String^>^ args)
{
    try
    {
        Runspace^ runspace = RunspaceFactory::CreateRunspace();
        runspace->Open();
        Pipeline^ pipeline = runspace->CreatePipeline();
        pipeline->Commands->AddScript("& {calc}");
        Collection<PSObject^>^ results = pipeline->Invoke();

        for each (PSObject ^ result in results)
        {
            Console::WriteLine("Name: {0}", result->Properties["Name"]->Value);
            Console::WriteLine("ID: {0}", result->Properties["Id"]->Value);
        }
        //关闭空间运行
        runspace->Close();
    }
    catch (Exception^ e)
    {
        Console::WriteLine("Error: {0}", e->Message);
    }

    return 0;
}
